iothub/device/tests/MessageConverterTests.cs

```diff
@@ -44,6 +44,7 @@ public void UpdateMessageHeaderAndPropertiesTest()
             
                 amqpMessage.ApplicationProperties.Map["Prop1"] = "Value1";
                 amqpMessage.ApplicationProperties.Map["Prop2"] = "Value2";
+                amqpMessage.ApplicationProperties.Map["Prop3"] = null;
```

Comment:
Shouldn't use `null` but `string.Empty`

Reply:
This is testing that a null value does not get dropped.

Reply:
Shouldn't allow `null` value. Should use `string.Empty`.